cdap-data-fabric/src/main/java/co/cask/cdap/data2/dataset2/lib/table/leveldb/LevelDBTableService.java

```diff
@@ -176,47 +186,39 @@ public void ensureTableExists(String tableName) throws IOException {
       synchronized (tables) {
         db = tables.get(tableName);
         if (db == null) {
-          createTable(tableName);
+          db = getDB(tableName, true);
+          tables.put(tableName, db);
         }
       }
     }
   }
 
-  private DB openTable(String tableName) throws IOException {
+  private DB getDB(String tableName, boolean createIfMissing) throws IOException {
     String dbPath = getDBPath(basePath, tableName);
 
     Options options = new Options();
-    options.createIfMissing(false);
+    options.createIfMissing(createIfMissing);
```

Comment:
Based on the comment below, shouldn't we keep this as `false` always?

Reply:
For the case that we pass in false, we do a directory exists check to ensure that it does not encounter the scenario described in the comments.
The scenario described in the comment doesn't occur when passing in `true`. We pass in `true` when creating a table, or ensuring that it exists.